flink-metrics/flink-metrics-datadog/src/main/java/org/apache/flink/metrics/datadog/DatadogHttpClient.java

```diff
@@ -48,6 +48,8 @@
 	private static final int TIMEOUT = 3;
 	private static final ObjectMapper MAPPER = new ObjectMapper();
 
+	static boolean skipApiKeyValidation = false;
```

Comment:
shall we add a comment to mention that this option is for testing?

Reply:
or `@VisibleForTesting`

Reply:
What about introducing a constructor `DatadogHttpClient(dgApiKey, dgProxyHost, dgProxyPort, boolean validateApiKey)` instead of this static field? It is also not perfect but I think it is a bit better than having this static field.

Reply:
I would go with that solution once factories are in place; as it stands we would need 2 constructors; one with the arguments, one no-arg constructor for reflection purposes (which would be detected as being unused), and I don't want to have to watch out for someone removing the constructor because he thought it's not used anyway, or migrate tests towards the new "convenient" constructor at which point we lose test coverage of runtime paths.

I know it's super icky and Ideally we get rid of the reflection approach soon.

Reply:
ooooooooooooooooooooooooooohhhhh....this is the _client_...yeah then we should totally do that.